lib/services/elasticsearch.js

```diff
@@ -153,519 +166,479 @@ class ElasticSearch extends Service {
 
         // Add metadata
         esRequest.body._kuzzle_info = {
-          author: this._getUserId(request),
+          author: String(userId),
           createdAt: Date.now(),
           updatedAt: null,
-          updater: null,
-          active: true,
-          deletedAt: null
+          updater: null
         };
 
-        if (esRequest.id) {
-          // Check if the document exists and has not been deleted (active: false)
-          return this.client.get(
-            {index: esRequest.index,
-              type: esRequest.type,
-              id: esRequest.id}
-          )
-            .then(result => {
-              if (result._source._kuzzle_info &&
-                !result._source._kuzzle_info.active) {
-                // The document is inactive, we replace it and masquerade the result as a creation
-                return this.client.index(esRequest)
-                  .then(res => {
-                    res.result = 'created';
-                    res.created = true;
-
-                    return this.refreshIndexIfNeeded(
-                      esRequest,
-                      _.extend(res, {_source: request.input.body}));
-                  })
-                  .catch(error => Bluebird.reject(
-                    this.esWrapper.formatESError(error))
-                  );
-              }
+        const promise = esRequest.id
+          ? this.client.exists({ index: esRequest.index, id: esRequest.id })
+          : Bluebird.resolve({ body: false });
 
-              // The document exits and is active, we reject to prevent the creation
-              return this.reject('document_already_exists', esRequest.id);
-            })
-            // Pitfall of all previous rejections
-            .catch(err => {
-              if (err.displayName === 'NotFound') {
-                // The document doesn't exist, we create it
-                return this.client.create(esRequest)
-                  .then(result => this.refreshIndexIfNeeded(
-                    esRequest,
-                    _.extend(result, {_source: request.input.body}))
-                  )
-                  .catch(error => Bluebird.reject(
-                    this.esWrapper.formatESError(error))
-                  );
-              }
-
-              // A "real" error occurred, we reject it
-              return Bluebird.reject(err);
-            });
+        return promise;
+      })
+      .then(({ body: exists }) => {
+        if (exists) {
+          return this.reject('document_already_exists', esRequest.id);
         }
 
-        return this.client.index(esRequest)
-          .then(result => this.refreshIndexIfNeeded(
-            esRequest,
-            _.extend(result, {_source: request.input.body}))
-          )
-          .catch(error => Bluebird.reject(
-            this.esWrapper.formatESError(error))
-          );
-      });
-  }
+        return this.client.index(esRequest);
+      })
+      .then(({ body }) => {
+        const apiResponse = {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Immediately return this expression instead of assigning it to the temporary variable "apiResponse". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1488)